check_init_and_setup_coincide.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+"""Check that the distribution and crosshair/__init__.py are in sync."""
+import subprocess
+import sys
+from typing import Optional, Dict
+
+import pkg_resources
+import email
+
+import crosshair
+
+
+def main() -> int:
```

Comment:
How would you feel about a dedicated metadata file in the package that setup.py can manually read/eval (e.g. that SO link you found has a suggestion along the lines of `open('crosshair/_metadata.py')`)

A little reluctant to suggest it, since you've put some amount of effort into this PR; but I just felt like the test was a fair amount of code to add, and if we can keep the metadata DRY, then we'll have a tiny bit less work to do at release time. WDYT?

Reply:
I tried the approach with the metadata in other projects (usually called `{project}_meta.py` like `icontract_meta.py`) and it worked for some time, but eventually the problems came around (either at the setup time, at the documentation build time, or on readthedocs servers). 

In the end, I had to denormalize after giving up the fight with setup.py, Sphinx, pip and readthedocs. I agree that it is clumsy to have such a long script test something so trivial, but at the moment that is the only way I know that works robustly.

Reply:
(See [this meta file](https://github.com/Parquery/gs-wrap/blob/master/gswrap_meta.py), for example. This worked, but the very same approach did not work in icontract on readthedocs.)